independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanRegistrar.java

```diff
@@ -34,6 +34,12 @@
             return configure(DotName.createSimple(beanClass.getName()));
         }
 
+        /**
+         * 
+         * @return a new stream instance
+         */
+        BeanStream beans();
```

Comment:
What exactly do I get here? Since when `BeanRegistrar`s are processed new beans are being added. That might be worth mentioning?

Reply:
Good point. You get all non-synthetic beans, I used the term `class-based beans` in the guide which is probably misleading.

Reply:
Hmm so you don't even get ones from producers, that might be a bit unexpected. Let's add some more javadoc here at least.

Reply:
Not really, "class-based" means all beans derived from the classes. I'll try to explain better...

Reply:
Actually, the synthetic beans are also contained. So it's basically "non-synthetic beans + all beans registered by extension before the stream is created".